src/db.ts

```diff
@@ -157,7 +157,7 @@ export class DB {
           value = value ? 1 : 0;
         // fall through
         case "number":
-          if (Math.floor(value) === value) {
+          if (Number.isSafeInteger(value) && !Object.is(value, -0)) {
```

Comment:
Can we remove the `Object.is(value, -0)` check?

You are right, that it will preserve the IEEE `-0` through a `SELECT ?` type query. But if you first store the values into a database table, and then retrieve them, the `-0` becomes a `+0` again (SQLite will be doing some serialisation, which does not preserve the difference).

Since the `-0` is not preserved through storing to a table, I don't think it makes sense to try preserving it through a `SELECT ?`, especially since you might do some arithmetic on integers and get a `-0` 'integer', which you actually want to keep as an integer 😅 

(Example of how `-0` is not persisted through table:)
```JavaScript
import { DB } from "./mod.ts";

const db = new DB();
db.query("CREATE TABLE numbers (id INTEGER PRIMARY KEY AUTOINCREMENT, num REAL)");

const val = -0;
db.query("INSERT INTO numbers (num) VALUES (?)", [val]);
const [[outVal]] = db.query("SELECT num FROM numbers");

console.log(`In: ${val}; Out: ${outVal}`);
console.log(`= ${val === outVal} : is ${Object.is(val, outVal)}`);
```

Reply:
Great point, done.

I wasn't thinking about `-0` at first, but noticed while writing the tests it seemed to be preserved when `SELECT`ing it, so I decided to include it. I didn't think to check that it was actually preserved when persisted to a real database. Not much point adding this special case if it's not.